src/fmnoise/coldbrew.clj

```diff
@@ -181,11 +195,14 @@
                               :else 3))
         fdefn (filter some? (list name docstring args prepost))]
     `(do
-      (def ^:private ~fname
-         (cached
-          (with-meta
-            (fn [~@cache-key]
-              ~@body)
-            ~cache-options)))
-      (defn ~@fdefn (~fname ~@cache-key))
-      (vary-meta ~name merge ~(meta name)))))
+       (def ^:private ~fname (cached
+                              (with-meta
+                                (fn [~@cache-key]
+                                  ~@body)
+                                ~cache-options)))
+       (defn ~@fdefn (~fname ~@cache-key))
+       (alter-meta! (var ~name) merge (meta ~fname) ~(meta name)))))
+
+(defmacro defcached-cache [defcache-fn]
+  `(cached-cache (var ~defcache-fn)))
```

Comment:
this will not work, because cache is attached to meta of the private function with name generated using gensym
so, we should still find a way to access cache